src/rules/function-blacklist/index.js

```diff
@@ -0,0 +1,41 @@
+import { isString } from "lodash"
+import { vendor } from "postcss"
+import valueParser from "postcss-value-parser"
+
+import {
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "function-blacklist"
+
+export const messages = ruleMessages(ruleName, {
+  rejected: (name) => `Unexpected function "${name}"`,
+})
+
+export default function (blacklist) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: blacklist,
+      possible: [isString],
+    })
+    if (!validOptions) { return }
+    root.walkDecls(decl => {
+      const { value } = decl
+      valueParser(value).walk(function (node) {
+        if (node.type === "function" && blacklist.indexOf(vendor.unprefixed(node.value)) !== -1) {
+          const charsBeforeColon = decl.toString().indexOf(":")
+          const charsAfterColon = decl.raw("between").length - decl.raw("between").indexOf(":")
```

Comment:
I originally had it that way, but `decl.raw("between")` returns the characters _before_ and after the colon. So, given

```css
a { transform : scale(1); }
```

`charsBeforeColon + decl.raw("between").length` would trip up by one place. Does that sound about right to you?

I know it looks messy. There has to be a neater way of get the index of `node.value` within a declaration.